src/main/java/com/board/config/boardconfiggui/LoadDataController.java

```diff
@@ -100,29 +136,31 @@ protected void onButtonClick(Event event) {
             switch (sourceNode.getUserData().toString()){
                 case Constants.XML_BUTTON_TYPE:
                     xmlPathField.setText(selectedFolder.getAbsolutePath());
-                    this.xmlFolderPath = xmlPathField.getText();
+                    configPathsModel.setXmlPath(xmlPathField.getText());
                     break;
                 case Constants.REPO_BUTTON_TYPE:
                     repoPathField.setText(selectedFolder.getAbsolutePath());
-                    this.repositoryFolderPath = repoPathField.getText();
+                    configPathsModel.setRepoPath(repoPathField.getText());
                     break;
                 case Constants.TOOL_CHAIN_BUTTON_TYPE:
                     toolChainPathField.setText(selectedFolder.getAbsolutePath());
-                    this.toolChainFolderPath = toolChainPathField.getText();
+                    configPathsModel.setToolchainPath(toolChainPathField.getText());
                     break;
                 case Constants.OUTPUT_BUTTON_TYPE:
                     outputPathField.setText(selectedFolder.getAbsolutePath());
-                    this.outputLocationFolderPath = outputPathField.getText();
+                    configPathsModel.setOutputPath(outputPathField.getText());
                     break;
                 default:
                     break;
             }
         }
 
-        boolean allFields = !xmlPathField.getText().isEmpty() && !repoPathField.getText().isEmpty() && !toolChainPathField.getText().isEmpty() && !outputPathField.getText().isEmpty();
-        if (allFields){
-            submitBtn.setDisable(false);
-        }
+        boolean allFields = StringUtils.isNotEmpty(configPathsModel.getXmlPath()) &&
```

Comment:
Write a method inside configpathsmodel class to validate all these fields like below,

public boolean isValid(){
 //check for path availability and return the response from here.